### PR TITLE
Fix overlap source target 296

### PR DESCRIPTION
**Pull request by ericzolf**
So shortly before release, I only dared to come with a minimal invasive workaround, I'll fix it more correctly for the next release.

The underlying issue is that select_mirror can't contain a selection object because it isn't pickable, but the function Select can't be called on a selection iterable, so it's a bit of a catch 22.

I also added a test case to the selection tests.

FIX: add workaround to avoid error when backup directory is under the source directory (see issue #296), there is a warning but the backup can succeed.
DEV: add test to backup to a repo within the source directory

### CODE REVIEW

**Comment by ericzolf** [comment]
Could someone review and squash merge? After that I could release...

**Comment by ottok** [comment]
Do you want to review this @heikkiorsila?

I skimmed through the code but I didn't quite understand it. I guess @heikkiorsila could at least test it?

File: src/rdiff_backup/Main.py [hunk 1]
```diff
@@ -584,15 +584,18 @@ def backup_warn_if_infinite_regress(rpin, rpout):
...
 
     relative_rpout_comps = tuple(rpout.path[len(rpin.path) + 1:].split(b'/'))
-    relative_rpout = rpin.new_index(relative_rpout_comps)
-    if not Globals.select_mirror.Select(relative_rpout):
-        return
+    relative_rpout = rpin.new_index(relative_rpout_comps)  # noqa: F841
+    # FIXME: this fails currently because the selection object isn't stored
+    #        but an iterable, the object not being pickable.
+    #        Related to issue #296
+    #if not Globals.select_mirror.Select(relative_rpout):  # noqa: E265
```
Comment (ikus060) [inline src/rdiff_backup/Main.py]: Tell me if I'm wrong: 
This was checking if the `relative_rpout` is part of the selection. If it was, it doesn't print the error messages.

**Comment by ikus060** [comment]
For me the changes it good.

**Comment by heikkiorsila** [comment]
I tested the change in my local setup. It seems to run correctly for our use case :) Thanks!

File: src/rdiff_backup/Main.py (hunk 1 above)
Comment (ericzolf) [reply src/rdiff_backup/Main.py]: More precisely, it was checking if Select was excluding it from the selection.

**Comment by ericzolf** [comment]
I'd like to merge and release, but it can wait until tomorrow.

**Comment by ericzolf** [comment]
Uuups. Forget it, sometimes the page doesn't refresh properly in GitHub.

**Pull request merged**